hs_core/hydroshare/hs_bagit.py

```diff
@@ -41,217 +50,104 @@ def delete_files_and_bag(resource):
+    zipfile = bdb.archive_bag(bagdir, "zip")
 
-def create_bag(resource):
-    """
-    Modified to implement the new bagit workflow. The previous workflow was to create a bag from
-    the current filesystem of the resource, then zip it up and add it to the resource. The new
-    workflow is to delegate bagit and zip-up operations to iRODS, specifically, by executing an
-    iRODS bagit rule to create bagit file hierarchy, followed by execution of an ibun command to
-    zip up the bagit file hierarchy which is done asychronously as a celery task. This function
-    only creates all bag files under resource collection in iRODS and set bag_modified iRODS AVU
-    metadata to true so that the bag will be created or recreated on demand when it is being
-    downloaded asychronously by a celery task.
+    # save the zipped bag to iRODS for retrieval upon download request
+    istorage = resource.get_irods_storage()
+    bag_full_name = 'bags/{res_id}.zip'.format(res_id=resource.short_id)
+    irods_dest_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+    destbagfile = os.path.join(irods_dest_prefix, bag_full_name)
 
-    Parameters:
-    :param resource: (subclass of AbstractResource) A resource to create a bag for.
-    :return: the hs_core.models.Bags instance associated with the new bag.
-    """
-    create_bag_files(resource)
+    # delete any existing bag files
+    if (istorage.exists(destbagfile)):
+        istorage.delete(destbagfile)
 
-    # set bag_modified-true AVU pair for on-demand bagging.to indicate the resource bag needs to be
-    # created when user clicks on download button
-    resource.setAVU("bag_modified", True)
+    istorage.saveFile(zipfile, destbagfile, True)
+
+    # set bag_modified to false for a newly created bag
+    path = resource.root_path
+    istorage.setAVU(path, "bag_modified", "false")
 
     # delete if there exists any bags for the resource
     resource.bags.all().delete()
+
     # link the zipped bag file in IRODS via bag_url for bag downloading
     b = Bags.objects.create(
         content_object=resource.baseresource,
         timestamp=resource.updated
     )
 
+
     return b
 
 
-def read_bag(bag_path):
-    """
-    :param bag_path:
-    :return:
-    """
+def get_remote_file_manifest(resource):
+    json_data = ''
+    data_list = []
+    for f in ResourceFile.objects.filter(object_id=resource.id):
+        data = {}
+        irods_file_name = resource.short_id + "/data/contents/" + f.file_name
+        irods_dest_prefix = "/" + settings.IRODS_ZONE + "/home/" + settings.IRODS_USERNAME
+        irods_server_prefix = settings.IRODS_HOST + ':' + settings.IRODS_PORT
+        irods_file_url = 'irods://' + irods_server_prefix +  irods_dest_prefix + "/" + irods_file_name
+        istorage = resource.get_irods_storage()
+
+        tmpdir = os.path.join(settings.TEMP_FILE_DIR, uuid4().hex)
+        tmpfile = os.path.join(tmpdir, f.file_name)
+
+        os.makedirs(tmpdir)
```

Comment:
I see the calling routine makes tmpdir as well with a different randomly generated string by uuid4().hex call. It'd be better  to make the temp dir as a parameter being passed in by the calling routine so that the same temp directory will be used. **Also you need to clear up temp directory every time after it is created**.